overlays/cabal-pkg-config.nix

```diff
@@ -0,0 +1,76 @@
+final: prev:
+{
+  # This is a wrapper for `cabal configure` use only.  It returns
+  # package names and versions based on lib/pkgconf-nixpkgs-map.nix.
+  # It works because cabal calls `--list-all` then passes all the
+  # packages returned by that to `--modversion`.
```

Comment:
I think it would be worth elaborating on this a bit - I think there's quite a bit of latent knowledge here!

Reply:
Yes more docs to come.  I'm still working out what to do about the implicit mapping of pkg-config names to nixpkgs names (we probably can't keep that).  Currently I have a script trawling through nixpkgs for `.pc` to make a more accurate mapping file.

Reply:
I've tried to expand this comment a bit.